core/src/main/java/org/commonjava/indy/core/content/DefaultContentManager.java

```diff
@@ -262,7 +262,15 @@ public Transfer retrieve( final ArtifactStore store, final String path, final Ev
             {
                 for ( final ArtifactStore member : members )
                 {
-                    item = doRetrieve( member, path, eventMetadata );
+                    try
+                    {
+                        item = doRetrieve( member, path, eventMetadata );
+                    }
+                    catch ( IndyWorkflowException e )
+                    {
+                        logger.error( "Failed to retrieve artifact from for path {} from {} in group {}, error is: {}",
+                                      path, member, store, e.getMessage() );
+                    }
                     if ( item != null )
```

Comment:
Hi, @ligangty 
I see about 5 methods call doRetrieve() in DefaultContentManager ,  I'm not sure where the 40X exception will throw, so why not we let the catch statement into doRetireve to ensure cover all the situations? like this one, https://github.com/yma88/indy/commit/ac67ed846e7ca2c2d88028bafb318e8ae3b449cb , Do you have other considers about this? maybe I miss something :(